webapp-django/crashstats/supersearch/tests/test_form_fields.py

```diff
@@ -61,3 +62,95 @@ def test_several_fields(self):
 
         eq_(field1.operator, '>')
         eq_(field2.operator, '<')
+
+    def test_several_fields_illogically_integerfield(self):
+        field = form_fields.IntegerField()
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>10', '<10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<10', '>10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<10', '>=10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<=10', '>10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<10', '<10']
+        )
+
+    def test_several_fields_illogically_datetimefield(self):
+        field = form_fields.DateTimeField()
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>2016-08-10', '<2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<2016-08-10', '<2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>=2016-08-10', '<2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>2016-08-10', '<=2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>=2016-08-10', '<=2016-08-09']
+        )
+        # but note, this should work!
+        field.clean(['>=2016-08-10', '<=2016-08-10'])
+
+        # any use of the equal sign and a less or greater than
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['=2016-08-10', '<2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['=2016-08-10', '>2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>2016-08-10', '=2016-08-10']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<2016-08-10', '=2016-08-10']
+        )
+
+        # more than two fields
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['>2016-08-01', '<=2016-08-02', '>=2016-08-02']
+        )
+        assert_raises(
+            ValidationError,
+            field.clean,
+            ['<2016-08-01', '<2016-08-02', '<2016-08-03']
```

Comment:
Yeah, tricky. 
If a human typed this into the UI in Super search and you leaned over her shoulder, would you say she's done it wrong? After all, it's like saying "I want something less than 10 and something less than 11 and something less than 12" which you "should" be saying "I want something less than 12".